### PR TITLE
PADV-926 change manage store to reduxin main and instructors pages

### PR DESCRIPTION
**Pull request by AuraAlba**
### Ticket
https://agile-jira.pearson.com/browse/PADV-926

### Description
Manage unique state with redux to shared state between components 

### Changes made
Define redux actions and reducer for instructors and main
Import redux store in other pages
Add and update tests

### How to test
Clone the Institution Portal MFE
Run npm install.
Run npm run start
Go to http://localhost:1980/

### CODE REVIEW

File: src/features/Instructors/AddInstructors/index.jsx [hunk 1]
```diff
@@ -1,40 +1,20 @@
-import React, { useState, useReducer, useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
```
Comment (anfbermudezme) [inline src/features/Instructors/AddInstructors/index.jsx:1]: We are not using this right now, Am I correct? Please add an inline comment showing that this module will change.  @AuraAlba

File: src/features/Students/StudentsPage/_test_/reducer.test.jsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import {
   UPDATE_CURRENT_PAGE,
 } from 'features/Students/actionTypes';
 import { RequestStatus } from 'features/constants';
-import reducer from 'features/Instructors/InstructorsPage/reducer';
+import reducer from 'features/Students/StudentsPage/reducer';
```
Comment (anfbermudezme) [inline src/features/Students/StudentsPage/_test_/reducer.test.jsx:8]: Why this change? @AuraAlba

File: src/features/Students/StudentsFilters/index.jsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ const initialState = {
 };
 
 const StudentsFilters = ({ resetPagination, fetchData, setFilters }) => {
```
Comment (anfbermudezme) [inline src/features/Students/StudentsFilters/index.jsx:36]: I suppose we will move this component to redux. Is that correct? @AuraAlba

File: src/features/Main/data/slice.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-param-reassign */
```
Comment (anfbermudezme) [inline src/features/Main/data/slice.js:1]: Why is it necessary? @AuraAlba

File: src/features/Students/StudentsPage/_test_/reducer.test.jsx (hunk 2 above)
Comment (AuraAlba) [reply src/features/Students/StudentsPage/_test_/reducer.test.jsx:8]: It was using the incorrect reducer, was the same definition and didn't give error so for the moment I import the correct reducer but that will be removed in the other ticket

File: src/features/Students/StudentsFilters/index.jsx (hunk 3 above)
Comment (AuraAlba) [reply src/features/Students/StudentsFilters/index.jsx:36]: Yes, in the other ticket the student and courses pages will be moved to redux

File: src/features/Main/data/slice.js (hunk 4 above)
Comment (AuraAlba) [reply src/features/Main/data/slice.js:1]: Eslint give this error because the reducer modify directly the property of the state, the good practice is make copy of the object and modify it but redux improve that so we can use it, only with reducer of redux

File: src/features/Instructors/AddInstructors/index.jsx (hunk 1 above)
Comment (AuraAlba) [reply src/features/Instructors/AddInstructors/index.jsx:3]: done

**Review by anfbermudezme — APPROVED** [review]
